pkg/controller/common/util.go

```diff
@@ -2088,3 +2094,30 @@ func SetPvcAllowedAnnotations(obj metav1.Object, pvc *corev1.PersistentVolumeCla
 		}
 	}
 }
+
+// ClaimMayExistBeforeDataVolume returns true if the PVC may exist before the DataVolume
+func ClaimMayExistBeforeDataVolume(c client.Client, pvc *corev1.PersistentVolumeClaim, dv *cdiv1.DataVolume) (bool, error) {
+	if IsUnbound(pvc) {
+		return false, nil
+	}
+	if ClaimIsPopulatedForDataVolume(pvc, dv) {
+		return true, nil
+	}
+	return ClaimAllowsAdoption(c, pvc)
+}
+
+// ClaimIsPopulatedForDataVolume returns true if the PVC is populated for the given DataVolume
+func ClaimIsPopulatedForDataVolume(pvc *corev1.PersistentVolumeClaim, dv *cdiv1.DataVolume) bool {
+	return pvc.Annotations[AnnPopulatedFor] == dv.Name
+}
+
+// ClaimAllowsAdoption returns true if the PVC may be adopted
+func ClaimAllowsAdoption(c client.Client, pvc *corev1.PersistentVolumeClaim) (bool, error) {
+	anno, ok := pvc.Annotations[AnnAllowClaimAdoption]
+	// if annotation exists, go with that regardless of featuregate
+	if ok {
+		val, _ := strconv.ParseBool(anno)
```

Comment:
I think maybe `allow` and `deny` as values makes more sense, with the name of the annotation not having allow in it.

Reply:
Not allowing adoption has been the expected behavior from the beginning, I think it makes sense to just treat the annotation existing as true. It also feels more consistent with other CDI annotations. Just my thoughts though.

Reply:
However, using booleans would allow explicitly disabling adoption even when the feature gate is enabled, right? That would make some testing scenarios more straightforward. In that case I think I prefer using `allow` and `deny`.

Reply:
Good point @alromeros